tests/foreman/ui/test_host.py

```diff
@@ -578,6 +580,102 @@ def test_negative_delete_primary_interface(self):
+            parameter
+
+        :CaseLevel: System
+        """
+        user_login = gen_string('alpha')
+        user_password = gen_string('alpha')
+        default_loc = entities.Location().search(
+            query={'search': 'name="{0}"'.format(DEFAULT_LOC)})[0]
+        role = entities.Role().create()
+        entities.Filter(
+            permission=entities.Permission(resource_type='Parameter').search(),
+            role=role,
+        ).create()
+        entities.Filter(
+            permission=entities.Permission(resource_type='Host').search(),
+            role=role,
+        ).create()
+        entities.User(
+            role=[role],
+            admin=False,
+            login=user_login,
+            password=user_password,
+            organization=[self.session_org],
+            location=[default_loc],
+            default_organization=self.session_org,
+        ).create()
+        param_name = gen_string('alpha')
+        host = entities.Host(
+            location=default_loc,
+            organization=self.session_org,
+            host_parameters_attributes=[
+                {'name': param_name, 'value': gen_string('alpha')}
+            ],
+        ).create()
+        with Session(self, user=user_login, password=user_password):
+            self.hosts.search_and_click(host.name)
+            self.hosts.click(locators['host.edit'])
+            self.hosts.remove_parameter(param_name)
+
+    @tier3
+    def test_negative_remove_parameter_non_admin_user(self):
+        """Attempt to remove host parameter as a non-admin user with
+        insufficient permissions
+
+        :BZ: 1317868
+
+        :id: 78fd230e-2ec4-4158-823b-ddbadd5e232f
+
+        :expectedresults: user with insufficient permissions is unable to
+            remove host parameter, 'Remove' link is not visible for him
+
+        :CaseLevel: System
+        """
+        user_login = gen_string('alpha')
+        user_password = gen_string('alpha')
+        default_loc = entities.Location().search(
+            query={'search': 'name="{0}"'.format(DEFAULT_LOC)})[0]
+        role = entities.Role().create()
+        entities.Filter(
+            permission=entities.Permission(name='view_params').search(),
+            role=role,
+        ).create()
+        entities.Filter(
+            permission=entities.Permission(resource_type='Host').search(),
+            role=role,
+        ).create()
+        entities.User(
+            role=[role],
+            admin=False,
+            login=user_login,
+            password=user_password,
+            organization=[self.session_org],
+            location=[default_loc],
+            default_organization=self.session_org,
+        ).create()
+        param_name = gen_string('alpha')
+        host = entities.Host(
+            location=default_loc,
+            organization=self.session_org,
+            host_parameters_attributes=[
+                {'name': param_name, 'value': gen_string('alpha')}
```

Comment:
Please move the param_name and param_value declarations to the beginning, with user_login and  user_password

Reply:
Thanks, will address. Except there's no need to store value in some `param_value` as we don't use it anywhere.